rnd/autogpt_builder/src/components/Flow.tsx

```diff
@@ -594,6 +681,22 @@ const FlowEditor: React.FC<{
           </ControlPanel>
         </ReactFlow>
       </div>
+      <RunnerInputUI
+        isOpen={isRunnerInputOpen}
+        onClose={() => setIsRunnerInputOpen(false)}
+        blockInputs={getBlockInputsAndOutputs().inputs}
+        onInputChange={handleInputChange}
+        onRun={() => {
+          setIsRunnerInputOpen(false);
+          requestSaveAndRun();
+        }}
+        isRunning={isRunning}
+      />
+      <RunnerOutputUI
+        isOpen={isRunnerOutputOpen}
+        onClose={() => setIsRunnerOutputOpen(false)}
+        blockOutputs={getBlockInputsAndOutputs().outputs}
+      />
```

Comment:
Does this UI Have to live exactly here, or can it be its own component with needed props passed down?

Reply:
likewise with all of its driving fucntionality

Reply:
driving reason is this file is already huge. I'd rather add one line than 100